sdk/compiler/damlc/lib/DA/Cli/Damlc/Command/MultiIde/Handlers.hs

```diff
@@ -254,10 +262,27 @@ clientMessageHandler miState unblock bs = do
               LSP.FcDeleted -> do
                 shutdownIdeByHome miState home
                 handleRemovedPackageOpenFiles miState home
+                atomically $ onSubIde_ miState home $ \ideData -> ideData {ideDataIsFullDamlYaml = False}
               LSP.FcCreated -> do
-                handleCreatedPackageOpenFiles miState home
-                rebootIdeByHome miState home    
-              LSP.FcChanged -> rebootIdeByHome miState home
+                isFullDamlYaml <- shouldHandleDamlYamlChange <$> T.readFile (unPackageHome home </> "daml.yaml")
+                if isFullDamlYaml
+                  then do
+                    handleCreatedPackageOpenFiles miState home
+                    rebootIdeByHome miState home
+                  else shutdownIdeByHome miState home
+                atomically $ onSubIde_ miState home $ \ideData -> ideData {ideDataIsFullDamlYaml = isFullDamlYaml}
+              LSP.FcChanged -> do
+                isFullDamlYaml <- shouldHandleDamlYamlChange <$> T.readFile (unPackageHome home </> "daml.yaml")
+                oldIdeData <- atomically $ onSubIde miState home $ \ideData -> (ideData {ideDataIsFullDamlYaml = isFullDamlYaml}, ideData)
+                if isFullDamlYaml
+                  then do
+                    when (not (ideDataIsFullDamlYaml oldIdeData) && Set.null (ideDataOpenFiles oldIdeData)) $
```

Comment:
We want to treat a daml.yaml file going from "only sdk-version" to a full package as though the daml.yaml was created. 
We track the previous time the file changed in order to know if it has become valid in this run.